opmd_viewer/addons/pic/lpa_diagnostics.py

```diff
@@ -17,4 +20,668 @@ def __init__(self, path_to_dir):
+        # (Modifies self.current_i and self.current_t)
+        self._find_output( t, iteration )
+        # Get particle data
+        x, y, ux, uy, w = self.get_particle(
+                                    var_list=['x', 'y', 'ux', 'uy', 'w'],
+                                    t=t, iteration=iteration,
+                                    species=species )
+        # Calculate the necessary RMS values
+        x *= 1.e-6
+        y *= 1.e-6
+        xsq = np.average( x ** 2, weights=w )
+        ysq = np.average( y ** 2, weights=w )
+        uxsq = np.average( ux ** 2, weights=w )
+        uysq = np.average( uy ** 2, weights=w )
+        xpx = np.average( x * ux, weights=w )
+        ypy = np.average( y * uy, weights=w )
+        # Calculate the beam emittances
+        emit_x = np.sqrt( xsq * uxsq - xpx ** 2 )
+        emit_y = np.sqrt( ysq * uysq - ypy ** 2 )
+        # Return the results
+        return( emit_x, emit_y )
+
+    def get_current( self, t=None, iteration=None, species=None, select=None,
+                     bins=100 ):
+        """
+        Calculate the electric current along the z-axis for selected particles.
+
+        Parameters
+        ----------
+         t : float (in seconds), optional
+            Time at which to obtain the data (if this does not correspond to
+            an available file, the last file before `t` will be used)
+            Either `t` or `iteration` should be given by the user.
+
+        iteration : int
+            The iteration at which to obtain the data
+            Either `t` or `iteration` should be given by the user.
+
+        species : string
+            Particle species to use for calculations
+
+        select : dict, optional
+            Either None or a dictionary of rules
+            to select the particles, of the form
+            'x' : [-4., 10.]   (Particles having x between -4 and 10 microns)
+            'z' : [0, 100] (Particles having x between 0 and 100 microns)
+
+        bins : float
+            Number of bins along the z-axis in which to calculate the current
+
+        Returns
+        -------
+        A tuple of arrays containig
+        - The current in each bin in Ampere
+        - The z positions of the bin edges
+        """
+        # Find the output that corresponds to the requested time/iteration
+        # (Modifies self.current_i and self.current_t)
+        self._find_output( t, iteration )
+        # Get particle data
+        z, uz, uy, ux, w, q = self.get_particle(
+                               var_list=['z', 'uz', 'uy', 'ux', 'w', 'charge'],
+                               t=t, iteration=iteration,
+                               species=species )
+        # Calculate Lorentz factor for all particles
+        gamma = np.sqrt(1 + ux ** 2 + uy ** 2 + uz ** 2)
+        # Length to be seperated in bins
+        len_z = np.max(z) - np.min(z)
+        # Define bins
+        bins_list = np.linspace(np.min(z), np.max(z), bins)
```

Comment:
`np.min(z)` and `np.max(z)` are calculated twice, which is not very efficient (z can be very long, as it represents the particles). What do you think about replacing
```python
 # Length to be seperated in bins
 len_z = np.max(z) - np.min(z)
 # Define bins
 bins_list = np.linspace(np.min(z), np.max(z), bins)
```
by
```python
 # Length to be separated in bins
zmin = np.min(z)
zmax = np.max(z)
 len_z = zmax - zmin
 # Define bins
 bins_list = np.linspace(zmin, zmax, bins)